services/controller/src/plz/controller/db_storage.py

```diff
@@ -0,0 +1,60 @@
+import json
+import logging
+from abc import ABC, abstractmethod
+from typing import Set
+
+from redis import StrictRedis
+
+log = logging.getLogger(__name__)
+
+
+class DBStorage(ABC):
+    @abstractmethod
+    def store_start_metadata(self, execution_id: str, start_metadata: dict) \
+            -> None:
+        pass
+
+    def retrieve_start_metadata(self, execution_id: str) -> dict:
+        pass
+
+    def add_execution_id_for_user_and_project(
+            self, user: str, project: str, execution_id: str) -> None:
+        pass
+
+    def retrieve_execution_ids_for_user_and_project(
+            self, user: str, project: str) -> Set[str]:
+        pass
+
+
+class _RedisDBStorage(DBStorage):
```

Comment:
Move this to another file and make it public. Nothing needs to import it apart from _configuration.py_.

Reply:
Done